src/pages/main/MainPage.tsx

```diff
@@ -84,6 +77,25 @@ export default function MainPage() {
     setRecentProjectHandlers,
   } = useRecentProjects();
   const { filesReferenceData, htmlReferenceData } = useReferenceData();
+  const {
+    monacoEditorRef,
+    setMonacoEditorRef,
+    iframeRefRef,
+    setIframeRefRef,
+    isContentProgrammaticallyChanged,
+    setIsContentProgrammaticallyChanged,
+    isCodeTyping,
+    setIsCodeTyping,
+  } = useReferneces();
+  const { invalidFileNodes, addInvalidFileNodes, removeInvalidFileNodes } =
+    useInvalidFileNodes(); // invalid - can't do any actions on the nodes
+  const {
+    temporaryFileNodes,
+    addTemporaryFileNodes,
+    removeTemporaryFileNodes,
+  } = useTemporaryFileNodes(); // temporary - don't display the nodes
```

Comment:
These comments can be directly added inside the hooks so the person can directly open the hook to get more details. This way the comments is not required to be written everywhere the hook is used

Reply:
okay.